src/components/pinnable.js

```diff
@@ -0,0 +1,90 @@
+AFRAME.registerComponent("pinnable", {
+  schema: {
+    pinned: { default: false }
+  },
+
+  init() {
+    this._apply = this._apply.bind(this);
```

Comment:
Apply is a bit of an overlaoded function name.. I would call something like `applyState` or `updateProperties`